### PR TITLE
Validate invoice before locking it

### LINKED ISSUE #111: Validate invoice on lock
When we lock an invoice validate that all fields are ready for the invoice to be sent to Fakturabank if not - do not lock.

### PR DESCRIPTION
**Pull request by mladjo2505**
Closes #111

### CODE REVIEW

**Comment by stoivo** [comment]
I am getting `get_accountplan_object(): Mangler AccountplanID` every time I save.

**Comment by mladjo2505** [comment]
@stoivo I will check this out, I've fixed the same thing on PR #80 for invoice incoming, where the problem was that the `AccountPlanID` was not passed to the `get_accountplan_object()` method.
I suspect the same thing is happening here just need to find where.

**Comment by mladjo2505** [comment]
Can't reproduce the `get_accountplan_object(): Mangler AccountplanID` for invoice outgoing.
Works fine at my place, tried all I could think should reproduce it. Also in invoice.class.php i found all occurrences of that function call and they all get the correct params.

**Comment by stoivo** [comment]
Not me either. I guess this is good then.

**Comment by stoivo** [comment]
We should validate Fakturadato and Forfallsdato for 0000-00-00
We should validate all fields under Leveringsadresse (Adresse, Postnummer, Poststed, Land)
Also we want to to show the validation messages under the lock button all the time (still keep the old one). If there are any invalid stuff the lock and fakturabank buttons should be disabled.

**Comment by stoivo** [comment]
The warning messages on top of page are printed twice. This didn't happen before.

File: modules/invoice/view/record.inc [hunk 1]
```diff
@@ -124,4 +113,29 @@ elseif($_lib['input']->getProperty('action_save_internal')) {
...
+  $invoice->init(array());
+  list($ready_to_send_to_fb, $error_messages) = $invoice->fakturabank_send_precheck();
+}
+
+if($_lib['input']->getProperty('action_invoice_fakturabanksend') || $_lib['input']->getProperty('action_invoice_lock')) {
+  foreach($error_messages as $error_message) $_lib['message']->add(array('message' => $error_message));
+}
+
+if($_lib['input']->getProperty('action_invoice_lock')) {
+    $invoice->init(array());
```
Comment (stoivo) [inline modules/invoice/view/record.inc]: I don't think we need this since we call init on line 119.

File: modules/invoice/view/record.inc [hunk 2]
```diff
@@ -124,4 +113,29 @@ elseif($_lib['input']->getProperty('action_save_internal')) {
...
+    if ($ready_to_send_to_fb) {
+      $invoice->lock();
+    }
+}
+elseif($_lib['input']->getProperty('action_invoice_fakturabanksend')) {
+    $invoice->init(array());
```
Comment (stoivo) [inline modules/invoice/view/record.inc]: I don't think we need this since we call init on line 119.

**Comment by mladjo2505** [comment]
@stoivo The error duplication does not happen on my machine, you have to show me what happens.

**Comment by mladjo2505** [comment]
CustomerAccountPlanID is not loaded on first load of invoice edit page and so are the City, ZipCode and Name fields. Fix this.

**Comment by mladjo2505** [comment]
Fixed.

**Pull request merged**